qt/aqt/sync.py

```diff
@@ -135,13 +128,26 @@ def full_sync(
     elif out.required == out.FULL_UPLOAD:
         full_upload(mw, on_done)
     else:
-        choice = ask_user_to_decide_direction()
-        if choice == FullSyncChoice.UPLOAD:
-            full_upload(mw, on_done)
-        elif choice == FullSyncChoice.DOWNLOAD:
-            full_download(mw, on_done)
-        else:
-            on_done()
+        button_labels: list[str] = [
+            tr.sync_upload_to_ankiweb(),
+            tr.sync_download_from_ankiweb(),
+            tr.sync_cancel_button(),
+        ]
+
+        def callback(choice: str) -> None:
+            if choice == button_labels[0]:
+                full_upload(mw, on_done)
```

Comment:
I regret returning the button text in the old API - the index of the chosen button would have been sufficient, and it would allow us to avoid this hack:

```
        # work around KDE 'helpfully' adding accelerators to button text of Qt apps
        return txt.replace("&", "")
```

What do you think about returning the index instead in the new routine? The default_button arg could become an index then too.

Reply:
Agreed - this hack shouldn't be necessary with the new class since the return values are extracted from the originating argument values rather than the button text, but I can't think of a reason not to return the index except perhaps if the buttons are being dynamically generated, which seems unlikely.

Reply:
One thing that's annoying about using the index is that Yes returns 0, No returns 1, and Cancel returns the highest value, if the typical button order is used, but I suppose we'll have to live with that

Reply:
The calling code should be checking the index rather than truthiness, so that should be ok.

Reply:
I've left `ask_user` as using bools, which required compensating for this by not-ing the value, but I think in that function it would be confusing to return the index since it only asks a Yes/No question.

Reply:
That seems reasonable - thanks!